pkg/signup/signup_service.go

```diff
@@ -0,0 +1,105 @@
+package signup
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	crtapi "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/registration-service/pkg/kubeclient"
+
+	errors2 "github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/rest"
+)
+
+type SignupServiceConfiguration interface {
+	GetNamespace() string
+}
+
+type SignupService interface {
+	CreateUserSignup(ctx context.Context, username, userID string) (*crtapi.UserSignup, error)
+}
+
+type SignupServiceImpl struct {
+	Namespace   string
+	UserSignups kubeclient.UserSignupInterface
+}
+
+// NewSignupService creates a service object for performing user signup-related activities
+func NewSignupService(cfg SignupServiceConfiguration) (SignupService, error) {
+	k8sConfig, err := rest.InClusterConfig()
```

Comment:
If we need to introduce more services which will talk to kube api server we will need to move this configuration initialization to main.go to avoid initializing it multiple times.
But it's OK to keep it here for now. Just something to keep in mind :)